[AG-13490] Fix(row-sorting): maintain group order

sdwvit opened the pull request:
Corrected icon directions and refactored sorting code to support `groupMaintainOrder` feature

Fix [AG-13490](https://ag-grid.atlassian.net/browse/AG-13490)

seanlandsman committed:
DASH-11 Auto grid bump and publish trigger

seanlandsman committed:
DASH-11 Auto grid bump and publish trigger

seanlandsman committed:
DASH-11 Auto grid bump and publish trigger

sdwvit committed:
Fix icon direction

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
SortType can't be null

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
- SortType can't be null
- getSortDef() may return null

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
Maintain group order during sorting interactions

 - Introduced _hasUserInteractedWithSort flag to track user interactions
 - Renamed setState to initState for clarity in initialization
 - Updated setSortDef to mark interaction only if not initial
 - Adjusted sortStage logic to check user interaction for sort removal

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
Fix sort interaction handling and maintain group order

 - Update user interaction flag to reflect initial sort state correctly
 - Ensure sort definition is set without null checks for consistency
 - Adjust handling of undefined sort to support groupMaintainOrder feature

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
- Rename _hasUserInteractedWithSort to _isSortExplicitlySet

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
- Ensure group order is preserved during sorting interactions

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
Merge tag 'latest' into sdwvit/fix/row-sorting/maintain-group-order/AG-13490

Latest commit to pass GitHub Actions workflow on latest branch.

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

# Conflicts:
#	.github/workflows/beta-publish.yml

sdwvit committed:
fix test

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

StephenCooper commented on .github/workflows/beta-publish.yml:
```diff
@@ -0,0 +1,59 @@
+name: Bump Beta Version
```
This seems like an accidental commit of this file?

StephenCooper committed:
Merge branch 'latest' into sdwvit/fix/row-sorting/maintain-group-order/AG-13490

StephenCooper committed:
Remove beta-publish to match latest

StephenCooper committed:
Remove unused import

sdwvit committed:
Refactor sorting state management to improve clarity and functionality

 - Rename _isSortExplicitlySet to _wasSortExplicitlyRemoved for clarity
 - Update setSortDef logic to correctly handle sort removal state
 - Adjust references to sorting state across multiple files
 - Enhance type validation for sorting direction and type

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

StephenCooper commented on documentation/ag-grid-docs/src/components/example-grid/config/colDefs.ts:
```diff
@@ -1,4 +1,4 @@
-import type { CellStyleFunc, ColDef, ColGroupDef, GridOptions } from 'ag-grid-community';
+import type { CellStyleFunc, ColDef, ColGroupDef, GridOptions, ValueFormatterParams } from 'ag-grid-community';
```
Is this unused?

sdwvit replied:
it is used

sdwvit committed:
Clarify behavior of group order preservation during sorting

 - Update documentation to specify that group order is not preserved if user resets sort direction

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>

sdwvit committed:
revert tests to stable version behaviour

Signed-off-by: Victor Musienko <8777372+sdwvit@users.noreply.github.com>
